File: MSCL_Examples/Wireless/C++/MSCL_Examples_C++/BaseStationBeacon/BaseStationBeacon.cpp
//MSCL Example: BaseStationBeacon
//	This example shows how to Enable and Disable the 
//	beacon (used for Synchronized Sampling) on a BaseStation.
//
//Updated: 10/01/2014

#include <iostream>
#include <chrono>
#include <thread>
using namespace std; 

#include "mscl/Communication/Connection.h"
#include "mscl/MicroStrain/Wireless/BaseStation.h"
#include "mscl/Exceptions.h"

int main(int argc, char **argv)
{
	//TODO: change these constants to match your setup
	const string COM_PORT = "COM3"; 
	 
	try
	{
		//create a BaseStation with the connection
		mscl::BaseStation base(mscl::Connection::Serial(COM_PORT));

		//make sure we can ping the base station
		if(!base.ping())
		{
			cout << "Failed to ping the Base Station" << endl;
		}

		cout << "Attempting to enable the beacon..." << endl;

		//enable the beacon on the Base Station using the PC time
		mscl::Timestamp beaconTime = base.enableBeacon();

		//if we got here, no exception was thrown, so enableBeacon was successful
		cout << "Successfully enabled the beacon on the Base Station" << endl;
		cout << "Beacon's initial Timestamp: " << beaconTime.str() << endl;

		cout << "Beacon is active" << endl << "Sleeping for 3 seconds..." << endl;
		
		std::this_thread::sleep_for(std::chrono::seconds(3));

		//disable the beacon on the Base Station
		base.disableBeacon();

		//if we got here, no exception was thrown, so disableBeacon was successful
		cout << "Successfully disabled the beacon on the Base Station" << endl;
	}
	catch(mscl::Error& e)
	{
		cout << "Error: " << e.what() << endl;
	}

	system("pause");
	return 0;
}

File: MSCL_Examples/Wireless/CSharp/MSCL_Examples_CSharp/BaseStationBeacon/BaseStationBeacon.cs
//MSCL Example: BaseStationBeacon
//	This example shows how to Enable and Disable the 
//	beacon (used for Synchronized Sampling) on a BaseStation
//
//Updated: 10/01/2014

//Note: If you are having 'PINVOKE' errors, please make sure the following is true:
//  -'MSCL_Managed' is added as a Refence for the project (add 'MSCL_Managed.dll')
//  -'MSCL.dll' is sitting next to the executable
//  -Your 'Platform Target' is set to x86

using System;
using System.Threading;

namespace BaseStationBeacon
{
    class Program
    {
        //TODO: change these constants to match your setup
        const String COM_PORT = "COM3";

        static void Main(string[] args)
        {
            try
            {
                //create a Serial Connection with the specified COM Port, default baud rate of 921600
                mscl.Connection connection = mscl.Connection.Serial(COM_PORT);

                //create a BaseStation with the connection
                mscl.BaseStation baseStation = new mscl.BaseStation(connection);

                //make sure we can ping the base station
                if (!baseStation.ping())
                {
                    Console.WriteLine("Failed to ping the Base Station");
                }

                Console.WriteLine("Attempting to enable the beacon...");

                //enable the beacon on the Base Station using the PC time
                mscl.Timestamp beaconTime = baseStation.enableBeacon();

                //if we got here, no exception was thrown, so enableBeacon was successful
                Console.WriteLine("Successfully enabled the beacon on the Base Station");
                Console.WriteLine("Beacon's initial Timestamp: " + beaconTime.ToString());

                Console.WriteLine("Beacon is active");
                Console.WriteLine("Sleeping for 3 seconds...");
                Thread.Sleep(3000);

                //disable the beacon on the Base Station
                baseStation.disableBeacon();

                //if we got here, no exception was thrown, so disableBeacon was successful
                Console.WriteLine("Successfully disabled the beacon on the Base Station");
            }
            catch (mscl.Error e)
            {
                Console.WriteLine("Error: " + e.Message);
            }

            Console.WriteLine("Press Enter to quit...");
            Console.Read();
        }
    }
}
